runtime/queries/hcl/highlights.scm

```diff
@@ -0,0 +1,102 @@
+; highlights.scm
+
+[
+  "!"
+  "\*"
+  "/"
+  "%"
+  "\+"
+  "-"
+  ">"
+  ">="
+  "<"
+  "<="
+  "=="
+  "!="
+  "&&"
+  "||"
+] @operator
+
+[
+  "{"
+  "}"
+  "["
+  "]"
+  "("
+  ")"
+] @punctuation.bracket
+
+[
+  "."
+  ".*"
+  ","
+  "[*]"
+] @punctuation.delimiter
+
+[
+  (ellipsis)
+  "\?"
+  "=>"
+] @punctuation.special
+
+[
+  ":"
+  "="
+] @none
+
+[
+  "for"
+  "endfor"
+  "in"
+] @repeat
+
+[ 
+  "if"
+  "else"
+  "endif"
+] @conditional
+
+[
+  (quoted_template_start) ; "
+  (quoted_template_end); "
+  (template_literal) ; non-interpolation/directive content
+] @string
+
+[
+  (heredoc_identifier) ; <<END
+  (heredoc_start) ; END
+] @punctuation.delimiter
+
+[
+  (template_interpolation_start) ; ${
+  (template_interpolation_end) ; }
+  (template_directive_start) ; %{
+  (template_directive_end) ; }
+  (strip_marker) ; ~
+] @punctuation.special
+
+(numeric_lit) @number
+(bool_lit) @boolean
+(null_lit) @constant
+(comment) @comment
+(identifier) @variable
```

Comment:
I think these are from the nvim-treesitter, yeah? nvim-treesitter's queries have backwards precedence (first rule top-to-bottom is considered last) while helix and tree-sitter-cli have it forwards (last rule top-to-bottom is considered last), so the identifier here is matching before the ones below on lines 93 and 94

if you move this 'default' rule for identifier below the other matches I think the queries will improve :+1:

Reply:
Was fighting with it for some time but didn't manage to make it work, I ended up using https://github.com/mitchellh/tree-sitter-hcl that I was able to setup but it seems to have less highlighting.